packages/features/eventtypes/repositories/eventTypeRepository.ts

```diff
@@ -1127,11 +1127,40 @@ export class EventTypeRepository {
   }
 
   async findFirstEventTypeId({ slug, teamId, userId }: { slug: string; teamId?: number; userId?: number }) {
+    // Use compound unique keys when available for optimal performance
+    // Note: teamId and userId are mutually exclusive - never both provided
```

Comment:
a follow-up idea:

would be great to do a refactor so that we break this function down into two: one dedicated for teamId and the other for userId